ledger/ledger-api-integration-tests/src/test/lib/scala/com/digitalasset/platform/apitesting/LedgerContext.scala

```diff
@@ -239,4 +242,47 @@ object LedgerContext {
     implicit def withResource(resource: Resource[SingleChannelContext]) =
       ResourceExtensions.MultiResource(resource)
   }
+
+
+  class MultiChannelContext(val mapping: Map[Party, LedgerContext],
+                            val defaultParty: Party,
+                            implicit val _esf: ExecutionSequencerFactory) extends LedgerContext {
+
+    override protected implicit def esf: ExecutionSequencerFactory = _esf
+
+    override def forParty(party: Option[Party]): LedgerContext = {
+      val p = party.getOrElse(defaultParty)
+      mapping.get(p) match {
+        case Some(lc) => lc
+        case None => throw new IllegalArgumentException("Unrecognised party: " + p + ", expecting one of: " + mapping.keySet.toString)
```

Comment:
using `require` would be more idiomatic in Scala

Reply:
How do I combine `require` with `mapping.get`? I do not see an elegant way to combine those two. I can do
```
mapping.getOrElse(p, require(false, "foobar"))
```
but it's bad.

I can do 
```
val x = mapping.get(p)
require(x.isDefined, "foobar")
x.value
```
which is also ugly.

Reply:
in such case i'd do something like:

```
require(mapping.contains(p), "Unrecognised party: ...")
val lc = mapping(p)
```

Reply:
Hm, I think `mapping.getOrElse(p, new IllegalArgumentException("..."))` is perfectly acceptable.

Reply:
why? `require` was introduced in Scala exactly to deal with such runtime assumptions

Reply:
for clarity, if the monadic usage of the `Option` was used further I would prefer the `getOrElse` way, but since we leave the Monad immediately I don't see why `require` is inferior here.

Reply:
All setups with `require` seem to require map lookup to happen twice. Ideally there should be a function equivalent to `def fail(msg) = require(false, msg)`. Then the `mapping.getOrElse(p, fail("..."))` would be the best option.

As not a Scala developer, I am leaning towards using `mapping.getOrElse(p, new IllegalArgumentException("..."))`.